### ISSUE TITLE
Generated `tsconfig.json` is not `json` compliant 

### ISSUE THREAD

**Issue by manolo** [vaadin/flow#7985]
When generating the `tsconfig.json` avoid any comment because IDE can complain about it.

https://vaadin.com/forum/thread/18187819/problem-vaadin-15-0-3-and-eclipse

**Comment by vlukashov** [comment]
If we really want to keep the comment AND make the file JSON-compliant, we could consider a hack like in
```json
{
  "__COMMENT__": [
    "This TypeScript configuration file is generated by vaadin-maven-plugin.",
    "This is needed for TypeScript compiler to compile your TypeScript code in the project.",
    "It is recommended to commit this file to the VCS.",
    "You might want to change the configurations to fit your preferences",
    "For more information about the configurations, please refer to ",
    "http://www.typescriptlang.org/docs/handbook/tsconfig-json.html",

    "NOTE: this is a comment",
    "But since JSON does not support //comments, it's added in this weird way."
  ],
  "compilerOptions": {
    "sourceMap": true,
```

**Comment by martinisraelsen** [comment]
This also happens in 14.1.27

**Comment by vlukashov** [comment]
> This also happens in 14.1.27

A `tsconfig.json` file in a Vaadin 14.1.27 app? That sounds very confusing to me as I am pretty confident Vaadin 14.x does not have built-in support for TypeScript. Could you please attach a reduced example project where you see this issue happen in Vaadin 14.1.27?

**Comment by martinisraelsen** [comment]
Hi @vlukashov,

I downloaded blank starter from Vaadin (14 LTS) and ran it with mvn jetty:run. A short while later the following shows up in eclipse.

![Screen Shot 2020-05-30 at 2 41 00 PM](https://user-images.githubusercontent.com/6271003/83337791-cd7a9600-a283-11ea-8522-367a39de495a.png)

**Comment by vlukashov** [comment]
Thanks! Now I understand the issue better. The screenshot shows that Eclipse complains about a file deep inside the `node_modules` folder. While this is not in direct control of Vaadin (it comes from a dependency and we do not generate this file nor can we modify it), I see how this becomes an annoying issue when you work with a Vaadin project in Eclipse.

I wonder, is there a way to configure Eclipse to not run any analysis on the contents of the `node_modules` folder.

**Comment by BrunoEberhard** [comment]
Yes you can configure eclipse not to run the json analysis for a project. Add exclude group and then add the rule.
(Be sure the rule says 'folder' and not 'file')

![exclude_node_modules](https://user-images.githubusercontent.com/10797221/84864463-fe760b80-b076-11ea-9812-7028b3b0ae39.PNG)

**Comment by martinisraelsen** [comment]
Thanks @vlukashov and @BrunoEberhard. Best tip ever!

**Comment by Obiwanbde** [comment]
Exactly the same happens for Vaadin 14.3.0 and with latest Eclipse even for the my-todo starter. 

And thanks to @BrunoEberhard , JSON Validator Exclude rule for npm_modules folder clears the errors.

**Comment by pleku** [comment]
I've made a PR for documentation on "using Eclipse" about disabling JSON syntax for `node_modules`. This issue can be still considered for the generated `tsconfig.json` file. But for 14 at least there is nothing else we can do.
